functions/zoom-log-notifications.py

```diff
@@ -1,46 +1,26 @@
 
 import json
-import zlib
+import gzip
 import base64
 import boto3
-from botocore.exceptions import ClientError
 from os import getenv as env
 
 import logging
 from common import setup_logging
 logger = logging.getLogger()
 
-STACK_NAME = env('STACK_NAME')
 SNS_TOPIC_ARN = env('SNS_TOPIC_ARN')
-LOG_RETENTION_DAYS = 180
-SUBSCRIPTION_FILTER_PATTERN = '{ $.level = "ERROR" }'
 
-cwlogs = boto3.client('logs')
-lmbda = boto3.client('lambda')
 sns = boto3.client('sns')
 
 
 @setup_logging
 def handler(event, context):
 
-    # we were triggered by a CreateLogGroup event from cloudtrail
-    if 'source' in event and event['source'] == 'aws.logs':
-        create_log_subscription(event, context)
-
-    # we were triggered by the log group subscription
-    elif 'awslogs' in event:
-        events2sns(event, context)
-
-
-def events2sns(event, context):
-
         logger.info("processing subscribed events")
 
         # log event data comes base64 encoded & gzipped
-        raw_data = zlib.decompress(
-            base64.b64decode(event['awslogs']['data']),
-            wbits=16
-        )
+        raw_data = base64.b64decode(gzip.decompress(event['awslogs']['data']))
```

Comment:
I couldn't get the zlib method to work in a reverse test context, so switched to gzip which is more straightforward anyway.